tests/integration/server/meeting/meeting.controller.test.ts

```diff
@@ -0,0 +1,1239 @@
+      });
+      context('Updating a course instance', function () {
+        let testCourseInstance: CourseInstance;
+        let newMeetings: MeetingRequestDTO[];
+        context('When there are no existing meetings', function () {
+          beforeEach(async function () {
+            const getMeetinglessCourse = courseInstanceRepository
+              .createQueryBuilder('ci')
+              .where('m.id IS NULL')
+              .leftJoinAndSelect(Meeting, 'm', 'm."courseInstanceId" = ci.id');
+
+            (testCourseInstance = await getMeetinglessCourse.getOne());
+          });
+
+          context('adding new meetings', function () {
+            beforeEach(async function () {
+              newMeetings = [
+                dummy.mondayMeetingWithRoom,
+                dummy.wednesdayMeetingWithRoom,
+              ];
+
+              response = await request(api)
+                .put(`/api/meetings/${testCourseInstance.id}`)
+                .send({ meetings: newMeetings });
+              result = response.body;
+            });
+            it('Should save the meetings as part of the courseInstance', async function () {
+              const savedMeetings = await meetingRepository
+                .find({
+                  where: `"courseInstanceId"='${testCourseInstance.id}'`,
+                });
+              strictEqual(savedMeetings.length, newMeetings.length);
+              const mondayMeeting = savedMeetings
+                .find(({ day }) => day === DAY.MON);
+              const wednesdayMeeting = savedMeetings
+                .find(({ day }) => day === DAY.WED);
+              strictEqual(
+                mondayMeeting.startTime,
+                dummy.mondayMeetingWithRoom.startTime
+              );
+              strictEqual(
+                mondayMeeting.endTime,
+                dummy.mondayMeetingWithRoom.endTime
+              );
+              strictEqual(
+                wednesdayMeeting.startTime,
+                dummy.wednesdayMeetingWithRoom.startTime
+              );
+              strictEqual(
+                wednesdayMeeting.endTime,
+                dummy.wednesdayMeetingWithRoom.endTime
+              );
+            });
+            it('Should return the new meetings, formatted with room data', async function () {
+              const formattedMeetings = await meetingListingViewRepository
+                .createQueryBuilder('ci')
+                .where(
+                  '"courseInstanceId"=:ciid',
+                  { ciid: testCourseInstance.id }
+                )
+                .orderBy('day', 'ASC')
+                .addOrderBy('"startTime"', 'ASC')
+                .addOrderBy('"endTime"', 'ASC')
+                .leftJoinAndMapOne('ci.room', RoomListingView, 'r', 'r.id=ci."roomId"')
+                .getMany();
+              strictEqual(
+                JSON.stringify(result),
+                JSON.stringify(formattedMeetings)
+              );
```

Comment:
Why `JSON.stringify`, out of interest?

Reply:
It's just a quick way to resolve that fact that `result` is an array of plain objects while `formattedMeetings` is an array of `Meeting` objects

Reply:
Oh, i see - so it strips the fancy TypeORM typings from them. Makes sense.